### ISSUE TITLE
RAC `TagGroup` within a `GridList` item causes a crash in 1.3.0

### ISSUE THREAD

**Issue by Niznikr** [adobe/react-spectrum#6757]
### Provide a general summary of the issue here

`TagGroup` within a `GridList` item causes a crash in 1.3.0. Perhaps related to `Collection` or virtualization changes?

### 🤔 Expected Behavior?

```
<GridList>
  <GridListItem>
    Item
    <TagGroup>
      <TagList>
        <Tag>Tag</Tag>
      </TagList>
    </TagGroup>
  </GridListItem>
</GridList>
```
Renders fine in `1.3.0` like it did in `1.2.0`.

### 😯 Current Behavior

Crashes the page with multiple errors.

### 🖥️ Steps to Reproduce

https://stackblitz.com/edit/vitejs-vite-mj3bm2

### Version

1.3.0

### What browsers are you seeing the problem on?

Chrome

### What operating system are you using?

MacOS

**Comment by reidbarber** [comment]
Likely similar to https://github.com/adobe/react-spectrum/issues/6753, the `ListStateContext` may need to be cleared.

**Comment by devongovett** [comment]
This should be fixed in v1.3.1.

**Comment by Niznikr** [comment]
Thanks for the quick turnaround!
